core/state/database.go

```diff
@@ -75,10 +75,10 @@ type Trie interface {
 // concurrent use and retains cached trie nodes in memory. The pool is an optional
 // intermediate trie-node memory pool between the low level storage layer and the
 // high level trie abstraction.
-func NewDatabase(db ethdb.Database) Database {
+func NewDatabase(db ethdb.Database, cache int) Database {
```

Comment:
The PR would have touched less files, and the interface might have been more clean, if you had created a separate `NewDatabaseWithCache(db ..., cache int)` and kept `NewDatabase(db)` as is, with cache set to `0`....

Reply:
I did think about this. I just wanted to make sure I update all code paths. 0 is actually "infinite memory", not 0.